bgpd/bgp_nb_config.c

```diff
@@ -173,6 +173,16 @@ int bgp_router_destroy(struct nb_cb_destroy_args *args)
 	case NB_EV_APPLY:
 		bgp = nb_running_unset_entry(args->dnode);
 
+		/* make sure we withdraw any exported routes */
```

Comment:
My understanding is that that is the other direction - releasing routes imported from the vpn

Reply:
The whole shutdown procedure here stinks.  I really believe that both bgp_vpn_leak_unimport and this change beelong in bgp_delete, since that is the controlling function for deletion.  Not a blocking comment for this review

Reply:
Happy to follow up with this and tidy the whole thing up.

> The whole shutdown procedure here stinks. I really believe that both bgp_vpn_leak_unimport and this change beelong in bgp_delete, since that is the controlling function for deletion. Not a blocking comment for this review

happy to follow up with a tidy-up.

Reply:
moved this stuff to bgp_delete